src/libtls/tls_hkdf.c

```diff
@@ -626,6 +626,35 @@ METHOD(tls_hkdf_t, resume, bool,
 	return TRUE;
 }
 
+METHOD(tls_hkdf_t, binder, bool,
+	private_tls_hkdf_t *this, chunk_t seed, chunk_t *out)
+{
+	chunk_t binder_key, finished_key;
+
+	if (!generate_secret(this, TLS_HKDF_RES_BINDER, chunk_empty, &binder_key))
+	{
+		DBG1(DBG_TLS, "unable to derive binder key");
+		return FALSE;
+	}
+
+	if (!expand_label(this, binder_key, chunk_from_str("finished"), chunk_empty,
+		this->hasher->get_hash_size(this->hasher), &finished_key))
+	{
+		chunk_free(&binder_key);
```

Comment:
```suggestion
		chunk_clear(&binder_key);
```
Same below (also for `finished_key`).